src/core/ext/transport/chttp2/transport/chttp2_transport.c

```diff
@@ -1449,6 +1452,10 @@ static void perform_stream_op_locked(grpc_exec_ctx *exec_ctx, void *stream_op,
                       "stream was closed"),
             "send_trailing_metadata_finished");
       } else if (s->id != 0) {
+        if (!t->is_client) {
+          grpc_chttp2_mark_stream_closed(exec_ctx, t, s, true, false, false,
```

Comment:
This is the only point where there will be a real change. The stream will be considered closed here, and the server will be able to start another stream with the client after this.